js/main.js

```diff
@@ -33,5 +62,54 @@ const getRandomFloatNumber = (from, to, precision = 5) => {
   return Number((Math.random() * (to - from) + from).toFixed(precision));
 };
 
-getRandomIntegerNumber('55', 60);
-getRandomFloatNumber(2.25, 3.75, 2);
+const generateSubArray = (arr) => arr.filter(() => Math.random() < 0.5);
+
+const getFixLengthNumberString = (x, targetLength) => {
+  if (x.toString().length < targetLength) {
+    return `0${getFixLengthNumberString(x, targetLength - 1)}`;
+  }
+
+  return x;
+};
+
+const createOffer = () => {
+  const lat = getRandomFloatNumber(35.65, 35.7, 5);
+  const lng = getRandomFloatNumber(139.7, 139.8, 5);
+
+  return {
+    author: {
+      avatar: `img/avatars/user${getFixLengthNumberString(getRandomIntegerNumber(1, 10), 2)}.png`,
+    },
+    offer: {
+      title: 'Сдам',
+
+      address: `${lat}, ${lng}`,
+
+      price: getRandomIntegerNumber(1000, 100000),
+
+      type: OFFER_TYPES[getRandomIntegerNumber(0, OFFER_TYPES.length - 1)],
+
+      rooms: getRandomIntegerNumber(1, 10),
+
+      guests: getRandomIntegerNumber(1, 10),
+
+      checkin: OFFER_TYME_VALUES[getRandomIntegerNumber(0, OFFER_TYME_VALUES.length - 1)],
+
+      checkout: OFFER_TYME_VALUES[getRandomIntegerNumber(0, OFFER_TYME_VALUES.length - 1)],
+
+      features: generateSubArray(OFFER_FEATURES),
+
+      description: 'Здесь можно жить',
+
+      photos: generateSubArray(OFFER_PHOTOS),
```

Comment:
Не ошибка, но обычно поля в объекте друг от друга пустой строкой не отделяеют
```suggestion
      title: 'Сдам',
      address: `${lat}, ${lng}`,
      price: getRandomIntegerNumber(1000, 100000),
      ......
```

Reply:
Ага - это я с задания блок кода копировал, забыл пробелы убрать. Тоже в объектах не оставляю пустых строк между ключами.
Поправил.